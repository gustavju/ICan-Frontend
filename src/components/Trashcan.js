import React from 'react';
import TrashcanButton from './TrashcanButton';

const commands = [
    'empty',
    'addTrash',
    'startTrashFire'
];

const Trashcan = ({trashcan}) => (
    <div className="trashcan-card">
        <div className="trashcan-card__id">ID: {trashcan.trashcanId}</div>
        <div className="trashcan-card__trashlevel">Trashlevel: {trashcan.TrashcanHistoryEntry.trashLevel}%</div>
        <hr/>
        <div className="trashcan-card__temperature">Temperature: {trashcan.TrashcanHistoryEntry.temperature}Â°C</div>
        <div className="trashcan-card__text"><span>Location:</span> {trashcan.location.longitude}, {trashcan.location.latitude}</div>
        <div className="trashcan-card__text"><span>Last Emptied:</span> {trashcan.TrashcanHistoryEntry.lastEmptied}</div>
        <div className="trashcan-card__text"><span>Data updated:</span> {trashcan.TrashcanHistoryEntry.timestamp}</div>
        <div className="trashcan-card__button-container">
            {commands.map((command) => <TrashcanButton key={command} trashcanId={trashcan.trashcanId} command={command} />)}
        </div>
    </div>
);

export default Trashcan;
