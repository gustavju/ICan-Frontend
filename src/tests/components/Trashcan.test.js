import React from 'react';
import { shallow } from 'enzyme';
import Trashcan from '../../components/Trashcan';
import traschcans from '../fixtures/trashcans';

test('should render the Trashcan', () => {
    const wrapper = shallow(<Trashcan trashcan={traschcans[0]} />);
    expect(wrapper).toMatchSnapshot();
});