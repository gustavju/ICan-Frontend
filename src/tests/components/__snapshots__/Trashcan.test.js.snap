// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the Trashcan 1`] = `
<div
  className="trashcan-card"
>
  <div
    className="trashcan-card__id"
  >
    8c13a125-cb3f-4a81-8cc5-c64574b0761f
  </div>
  <CirclePercent
    percent="100.0"
  />
  <div
    className="trashcan-card__Conn"
  >
    Connected
  </div>
  <div
    className="trashcan-card__temperature"
  >
    Temperature: 
    128.0
    Â°C
  </div>
  <hr />
  <div
    className="trashcan-card__text"
  >
    <span>
      Status
    </span>
     
    ONFIRE
  </div>
  <div
    className="trashcan-card__text"
  >
    <span>
      Type
    </span>
     
    household
  </div>
  <div
    className="trashcan-card__text"
  >
    <span>
      Location
    </span>
     
    17.947002
    , 
    59.398546
  </div>
  <div
    className="trashcan-card__text"
  >
    <span>
      Last Emptied
    </span>
      
    Invalid date
  </div>
  <div
    className="trashcan-card__text"
  >
    <span>
      Data updated
    </span>
     
    Invalid date
  </div>
  <div
    className="trashcan-card__button-container"
  >
    <Connect(TrashcanButton)
      command="empty"
      key="empty"
      trashcanId="8c13a125-cb3f-4a81-8cc5-c64574b0761f"
    />
    <Connect(TrashcanButton)
      command="addTrash"
      key="addTrash"
      trashcanId="8c13a125-cb3f-4a81-8cc5-c64574b0761f"
    />
    <Connect(TrashcanButton)
      command="startTrashFire"
      key="startTrashFire"
      trashcanId="8c13a125-cb3f-4a81-8cc5-c64574b0761f"
    />
  </div>
</div>
`;
